docs: typescript login progress (#7378)
zitadel/zitadel: docs/docs/guides/integrate/login-ui/typescript-repo.mdx

---
title: Typescript Repository
sidebar_label: Typescript Repository
---

> To replace our current golang built login UI and showcase the use of our new resource and OIDC APIs, we've created the [Typescript Repository](https://github.com/zitadel/typescript).

Reviewer: Maybe we should mention the session API here, Do we have a new OIDC API?:

```suggestion
To replace our current golang built login UI and showcase the use of our new resource, session and OIDC APIs, we've created the [Typescript Repository](https://github.com/zitadel/typescript).
```

Author: > Maybe we should mention the session API here, Do we have a new OIDC API?:

I'd address this better with a drawing of the architecture of the new login.

> The Typescript Repository contains all TypeScript and JavaScript packages and applications you need to create your own ZITADEL Login UI.

Reviewer: ```suggestion
The typescript repository contains all TypeScript and JavaScript packages and applications you need to create your own Login UI using the ZITADEL APIs.
```

The repo implements the [OIDC standard](./oidc-standard)

## Included Apps And Packages

> - `login`: The login UI used by ZITADEL Cloud, powered by Next.js

Reviewer: Its not yet used by the ZITADEL Cloud, should we write something like that should be the login for ZITADEL cloud in the future? (As soon as we have the functionality ready)

```suggestion
- `login`: The login UI used by ZITADEL Cloud, powered by Next.js
```

Author: I've stated that it is the future login UI now 👍

- `@zitadel/server`: core components for establishing node client connection, grpc stub
- `@zitadel/client`: core components for establishing web client connection, grpc stub

> - `@zitadel/react`: shared React utilities and components built with tailwindcss

Reviewer: What components? Only for login? User Management? Can we add what it inlcudes?
```suggestion
- `@zitadel/react`: shared React utilities and components built with Tailwind CSS
```

> - `@zitadel/next`: shared Next.js utilities

Reviewer: What utilities are in there?

Author: > What utilities are in there?

This is under heavy development and could still change. This will combine react components with a server side request wrapper for example Login with IDPs.

Reviewer: Can we add what we currently have?

- `@zitadel/tsconfig`: shared `tsconfig.json`s used throughout the monorepo
- `eslint-config-zitadel`: ESLint preset

## Implemented features of OIDC Standard

> 

Reviewer: Can we add somehow that Login with username and password is already possible?

Author: I've added them under a new checkbox
